src/resonanceReconstruction/breitWigner/singleLevel/Apply/src/call.hpp

```diff
@@ -5,41 +5,43 @@ operator()( const ENDF::ResonanceRange& range,
 
   try {
 
-    EnergyRange energyRange{ range.EL() * electronVolts,
-                            range.EH() * electronVolts };
-    auto slbw = std::get< ENDF::resolved::SLBW >( range.parameters() );
+    const EnergyRange energyRange {
+      range.EL() * electronVolts,
+      range.EH() * electronVolts
+    };
+    const auto slbw = std::get< ENDF::resolved::SLBW >( range.parameters() );
+
+    return
+      ( range.NRO() && range.NAPS() == 0) ||
+      ( range.NRO() && range.NAPS() == 2) ||
+      (!range.NRO() && range.NAPS() == 0)
+    ? callback( build(
+          energyRange, slbw,
+          range.NAPS() == 0
+            ? channelRadius( slbw.lValues().front().AWRI() )
+            : radius( slbw.AP() ),
+          range.NRO()
+            ? radius( range.scatteringRadius().value() )
+            : radius( slbw.AP() )
+         )
+       )
+    : callback( build(
+          energyRange, slbw,
+          range.NRO()
+            ? radius( range.scatteringRadius().value() )
+            : radius( slbw.AP() )
+          )
+      );
+
+  } // try
```

Comment:
Why is this better than the `switch` statement? While the indentation you have done here helps readability, I'm not sure it's more readable than the original `switch`. Of course, if changing to this instead of the `switch` is what speeds things up, that is the primary goal.

Reply:
For many people, especially those who don't like ternary conditional as much as I do, my changes might be less readable. These changes are *precisely* what allowed me to speed up the code, however, apparently because fewer `callback(build(...))` calls trigger fewer expensive template instantiations elsewhere. If and when some bigger compilation performance problems are found, I anticipate that we could try putting back the original switch, if others find it more readable.